src/python/pants/engine/internals/graph.py

```diff
@@ -740,8 +763,8 @@ async def resolve_dependencies(
     )
 
     inference_request_types = union_membership.get(InferDependenciesRequest)
-    inferred = InferredDependencies()
-    if global_options.options.dependency_inference and inference_request_types:
+    inferred: Tuple[InferredDependencies, ...] = (InferredDependencies(),)
```

Comment:
Nit. It's fine to fix this in a followup, I know we're trying to land this for the release.
```suggestion
    inferred = ()
```